core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -917,6 +917,7 @@ void *cuda_resize_scratch_space(std::int64_t bytes, bool force_shrink) {
 
 void cuda_prefetch_pointer(const Cuda &space, const void *ptr, size_t bytes,
                            bool to_device) {
+  if ((ptr == nullptr) || (bytes == 0)) return;
```

Comment:
So if `!ptr` is `true`  I assume the `cudaPointerGetAttributes()` is the one that yields an error.
What was the issue with a zero size?

Reply:
the prefetch function doesn't work with zero size it seems.

Reply:
if you create `View<double*> a("A",0)`  `a.data()` is not nullptr but extent is zero. This seems to trip it up too. Might be that actually cudaPointerGetAttributes() is tripping up because technically `a.data()` is not a valid pointer. Its like `vector.end() == vector.begin()`